R/estimateParams.R

```diff
@@ -0,0 +1,55 @@
+#' Preparation of a Data Set to Compute Log-likelihood
+#'
+#' @param data (`data.frame`)\cr data frame containing entry and exit times of an illness-death model.
+#'   See [getOneClinicalTrial()] for details.
+#'
+#' @return This function returns a data set with one row per patient and transition, when the patient is at risk.
+#' @export
+#'
+#' @details
+#' The output data set contains the following columns:
+#' - id (`integer`): patient id.
+#' - from (`integer`): start event state.
+#' - to (`integer`): end event state.
+#' - trans (`integer`): transition (1, 2 or 3) identifier.
+#' - entry (`numeric`): time at which the patient begins to be at risk for the transition.
+#' - exit (`numeric`): time at which the patient ends to be at risk for the transition.
+#' - status (`logical`): event indicator for the transition.
+#'
+#' @examples
+#' transition <- exponential_transition(h01 = 1.2, h02 = 1.5, h12 = 1.6)
+#' simData <- getOneClinicalTrial(
+#'   nPat = c(30), transitionByArm = list(transition),
+#'   dropout = list(rate = 0.8, time = 12),
+#'   accrual = list(param = "time", value = 1)
+#' )
+#' prepareData(simData)
+#' @keywords internal
+prepareData <- function(data) {
+  assert_data_frame(data, min.cols = 9, max.cols = 11)
+  if (ncol(data) == 9) {
```

Comment:
do we really need this magic number 9 here? or could we check whether certain columns are not yet in the `data`?